libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteSession.kt

```diff
@@ -58,11 +60,23 @@ class ReplayRouteSession : MapboxNavigationObserver {
     private lateinit var replayRouteMapper: ReplayRouteMapper
     private var mapboxNavigation: MapboxNavigation? = null
     private var lastLocationEvent: ReplayEventUpdateLocation? = null
-    private var routesObserver: RoutesObserver? = null
-    private var currentRouteId: String? = null
+    private var currentRoute: NavigationRoute? = null
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        if (currentRoute != routeProgress.navigationRoute) {
```

Comment:
Can that be the case? I think this check can be done in `RoutesObserver`. It's not free and route progress updates are quite frequent.

Reply:
I agree with @dzinad. `RouteProgressObserver` is being added to forward the information about `currentRouteGrometryIndex`. Can we rather store this information as a class field, write to it in the `RouteProgressObserver` and observe it in the `RoutesObserver`.

Reply:
Yeah I was not attempting to be optimal yet, just getting it to work. 

I don't think it's a simple move to the RoutesObserver because the route change happens before the RouteProgress. We're looking for the first RouteProgress update that includes the new route.

Agree that this equals check is wasteful though, so I'll look into some tricks to move it

Reply:
actually, the route has an identifier which is perfect for low-cost-equality comparison

Reply:
Here are some logs captured when selecting an alternative route during active navigation. As you can see there is a onRoutesChanged, but it takes a delay (14.470-13.742=0.728) to receive the updated route progress.

```
09:17:10.443 onRouteProgressChanged 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0 != 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0
09:17:11.450 onRouteProgressChanged 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0 != 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0
09:17:12.448 onRouteProgressChanged 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0 != 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0
09:17:13.449 onRouteProgressChanged 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0 != 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0
09:17:13.742 onRoutesChanged 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1
09:17:14.470 onRouteProgressChanged 7V5qH7NIKxoAVclBJwKqpPS5nDUKxQCsqHWfPEhNh7j3HdTY0Kx92g==#0 != 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1
09:17:14.497 onRouteProgressChanged 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1 != 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1
09:17:15.506 onRouteProgressChanged 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1 != 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1
09:17:16.506 onRouteProgressChanged 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1 != 4_Smaoct9ISBxKmKQO3orhnHIHY-C5VKYpsVwQOMV1Lpr0mLXA9PHw==#1
```